### PR TITLE
feat(Close #5): design mongoose models

### LINKED ISSUE #5: design mongoose models

### PR DESCRIPTION
**Pull request by khanghuynh92**

### CODE REVIEW

File: models/action.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  },
+  entityType: {
+    type: String,
+    required: true,
+    enum: ['Post', 'Source']
+  },
+  createdAt: {
+    type: Date,
+    default: Date.now
+  },
```
Comment (phatpham9) [inline models/action.js]: change to `created` object that contains `at` with `Date` type & `by` with `ObjectId` type ref to `User`

Comment (phatpham9) [reply models/action.js]: also create `updated` object has same props as `created`

Comment (phatpham9) [reply models/action.js]: also create `deleted` object has same props as `created`

File: models/post.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      type: Number,
+      min: 0,
+      default: 0
+    }
+  },
+  createdAt: {
+    type: Date,
+    default: Date.now
+  },
+  isDeleted: {
```
Comment (phatpham9) [inline models/post.js]: same as above, add `created` `updated` & `deleted` objects

File: models/source.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  },
+  url: {
+    type: String,
+    trim: true,
+    required: true
+  },
+  createdAt: {
+    type: Date,
+    default: Date.now
+  },
```
Comment (phatpham9) [inline models/source.js]: add created updated & deleted objects

File: models/user.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+  },
+  token: {
+    type: String,
+    trim: true,
+    required: true
+  },
+  vendor: {
+    type: String,
+    trim: true
+  },
```
Comment (phatpham9) [inline models/user.js:22]: what is vendor?

File: models/user.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+  source: [
+    {
+      type: Schema.ObjectId,
+      ref: 'User'
```
Comment (phatpham9) [inline models/user.js]: why source ref user?

File: models/user.js [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+        trim: true
+      },
+      middleName: {
+        type: String,
+        trim: true
+      }
+    },
+    gender: {
+      type: String,
+      trim: true
```
Comment (phatpham9) [inline models/user.js]: should have enum

File: models/user.js [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+      {
+        type: String,
+        trim: true
+      }
+    ]
+  },
+  createdAt: {
+    type: Date,
+    default: Date.now
+  },
```
Comment (phatpham9) [inline models/user.js]: add created updated & deleted objects

File: models/user.js (hunk 4 above)
Comment (khanghuynh92) [reply models/user.js:22]: Vendor such as Facebook, Gmail, Twitter ..

File: models/user.js (hunk 5 above)
Comment (khanghuynh92) [reply models/user.js]: Updated!

Comment (khanghuynh92) [reply models/user.js]: source: [
    {
      type: Schema.ObjectId,
      ref: 'Source'
    }
  ]

**Review by phatpham9 — APPROVED** [review]

**Pull request merged**